### PR TITLE
feat: [46] - Add new user validation upon registration

### PR DESCRIPTION
**Pull request by Thevel0per**
### Purpose

We want to have registration endpoint validate new user params and return them in standardized format

### CODE REVIEW

File: app/contracts/users/create_contract.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    required(:role).filled(:string)
+  end
+
+  rule(:email) do
+    key.failure('is already in use') if User.exists?(email: value)
+    key.failure('must use valid format') unless value.match(Devise.email_regexp)
+  end
+
+  rule(:role) do
+    key.failure('must be valid role') unless %w[student teacher].include?(value)
```
Comment (tomczak-mateusz) [inline app/contracts/users/create_contract.rb:18]: "must be a non admin role" (?)

File: spec/contracts/users/create_contract_spec.rb [hunk 2]
```diff
@@ -0,0 +1,217 @@
+# frozen_string_literal: true
+
+RSpec.describe Users::CreateContract do
+  describe '#validate_with' do
+    let(:result) { subject.call(params) }
+
+    context 'when required values contain nils' do
```
Comment (tomczak-mateusz) [inline spec/contracts/users/create_contract_spec.rb:7]: maybe let's go with the context below as we do in other contract specs:
`context 'with missing required values' do `

File: spec/contracts/users/create_contract_spec.rb [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            password: ['must be filled'],
+            password_confirmation: ['must be filled'],
+            role: ['must be filled'],
+            full_name: ['must be filled']
+          }
+        )
+      end
+    end
+
+    context 'when required values are missing' do
```
Comment (tomczak-mateusz) [inline spec/contracts/users/create_contract_spec.rb:32]: maybe let's go with the context below as we do in other contract specs:
`context 'with missing required keys' do`

File: spec/contracts/users/create_contract_spec.rb [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        expect(result.success?).to be(false)
+        expect(result.errors.to_h).to eq(
+          {
+            role: ['must be valid role']
+          }
+        )
+      end
+    end
+
+    context 'when parameters are valid' do
```
Comment (tomczak-mateusz) [inline spec/contracts/users/create_contract_spec.rb:201]: maybe let's go with the context below as we do in other contract specs:
`context 'with valid params' do`

File: spec/contracts/users/create_contract_spec.rb (hunk 3 above)
Comment (Thevel0per) [reply spec/contracts/users/create_contract_spec.rb:32]: makes sense, my bad

**Review by tomczak-mateusz — APPROVED** [review]

**Pull request merged**